zookeeper/model.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+
+from tensorflow import keras
+from zookeeper import Component
+
+
+class Model(Component, ABC):
```

Comment:
How would this API work when using the [Keras model subclassing API](https://www.tensorflow.org/guide/keras/custom_layers_and_models#the_model_class)?

Reply:
Huh, that is a very good point that I had not considered. In theory a user could subclass both `keras.models.Model` and `Component` and it should all work nicely, but then the API would not be to call `build` on the class to return an actual model.

We could disambiguate the name by calling it `ModelFactory` or `ModelBuilder` or something along those lines.

It's certainly seems to me that the subclass API is much less popular than the functional API.

I'm not really sure what the right answer is for this.

Reply:
Would it still work, since we prevent people from overwriting `__init__`?

Reply:
Ah you're right, that wouldn't work because of overriding `__init__`